x/concentrated-liquidity/store.go

```diff
@@ -101,3 +102,71 @@ func ParseFullPositionFromBytes(key, value []byte) (types.FullPositionByOwnerRes
 		Liquidity:   positionValue.Liquidity,
 	}, nil
 }
+
+// ParseFullIncentiveRecordFromBz parses a IncentiveRecord from a byte array.
+// Returns a struct containing the denom and min uptime associated with the incentive record.
+// Returns an error if the byte array is empty.
+// Returns an error if fails to parse.
+func ParseIncentiveRecordFromBz(bz []byte) (incentiveRecord types.IncentiveRecord, err error) {
+	if len(bz) == 0 {
+		return types.IncentiveRecord{}, errors.New("incentive record not found")
+	}
+	err = proto.Unmarshal(bz, &incentiveRecord)
+	return incentiveRecord, err
+}
+
+// ParseIncentiveRecordFromBz parses an incentive record from a byte array.
+// Returns a struct containing the state associated with the incentive.
+// Returns an error if the byte array is empty.
+// Returns an error if fails to parse.
+func ParseFullIncentiveRecordFromBz(key []byte, value []byte) (incentiveRecord types.IncentiveRecord, err error) {
+	if len(key) == 0 {
+		return types.IncentiveRecord{}, errors.New("key not found")
+	}
+	if len(value) == 0 {
+		return types.IncentiveRecord{}, fmt.Errorf("value not found for key (%s)", value)
+	}
+
+	keyStr := string(key)
+
+	// These may include irrelevant parts of the prefix such as the module prefix.
+	incentiveRecordKeyComponents := strings.Split(keyStr, types.KeySeparator)
+
+	// We only care about the last 3 components, which are:
+	// - pool id
+	// - incentive denom
+	// - min uptime
+
+	relevantIncentiveKeyComponents := incentiveRecordKeyComponents[len(incentiveRecordKeyComponents)-3:]
```

Comment:
non-blocking Q: what do you think about grabbing these components from the end?

I did something similar here: https://github.com/osmosis-labs/osmosis/blob/792375a8ae5cf1b8d5ef41ac3f27bb1394abc5f4/x/concentrated-liquidity/store.go#L59-L64

However, I'm afraid that is not very extensible because any time we want to add a new proto field, this logic will break.

We might want to refactor this later to grab these components relative to the start as opposed to the end. Let me know what you think

Reply:
I figure even if we add a proto field it is unlikely we will add it to the key (in fact if it's in the key we probably won't make it a proto field), so I don't foresee us breaking these unless we do a major refactor to pool/position protos.

This is since I'm assuming these fields will always be sufficient to identify a position (or incentive record in my case), so we will likely leave the key unchanged. Does this align with your thinking or did you mean something else?

Reply:
Yeah, that makes sense - agreed